commands/azureCommands/delete-image.ts

```diff
@@ -0,0 +1,57 @@
+import { Registry } from "azure-arm-containerregistry/lib/models";
+import * as vscode from "vscode";
+import { dockerExplorerProvider } from '../../dockerExtension';
+import { UserCancelledError } from "../../explorer/deploy/wizard";
+import { AzureImageNode, AzureRepositoryNode } from '../../explorer/models/azureRegistryNodes';
+import { reporter } from '../../telemetry/telemetry';
+import * as acrTools from '../../utils/Azure/acrTools';
+import { AzureImage } from "../../utils/Azure/models/image";
+import { Repository } from "../../utils/Azure/models/repository";
+import { AzureUtilityManager } from '../../utils/azureUtilityManager';
+import * as quickPicks from '../utils/quick-pick-azure';
+
+const teleCmdId: string = 'vscode-docker.delete-ACR-Image';
+
+/** Function to delete an Azure hosted image
+ * @param context : if called through right click on AzureImageNode, the node object will be passed in. See azureRegistryNodes.ts for more info
+ */
+export async function deleteAzureImage(context?: AzureImageNode): Promise<void> {
+    if (!AzureUtilityManager.getInstance().waitForLogin()) {
+        vscode.window.showErrorMessage('You are not logged into Azure');
+        throw new Error('User is not logged into azure');
+    }
+    let registry: Registry;
+    let repoName: string;
+    let tag: string;
+
+    if (!context) {
+        registry = await quickPicks.quickPickACRRegistry();
+        const repository: Repository = await quickPicks.quickPickACRRepository(registry, 'Choose the Repository of the image you want to delete');
+        repoName = repository.name;
+        const image: AzureImage = await quickPicks.quickPickACRImage(repository, 'Choose the Image you want to delete');
```

Comment:
Please use sentence casing (https://vsuiguide.azurewebsites.net/topics/02%20Fonts%20and%20Formatting/02.02%20Text%20style.html).

Reply:
… same for any other prompts